presidio-analyzer/analyzer/recognizer_registry/recognizer_registry.py

```diff
@@ -63,38 +63,40 @@ def remove_recognizer(self, name):
         if not found:
             raise ValueError("Requested recognizer was not found")
 
-    def get_recognizers(self, entities=None, language=None):
+    def get_recognizers(self, language, entities=None, all_fields=False):
         """
         Returns a list of the recognizer, which supports the specified name and
-        language. if no language and entities are given, all the available
-        recognizers will be returned
+        language.
         :param entities: the requested entities
         :param language: the requested language
+        :param all_fields: a flag to return all fields of a requested language.
         :return: A list of the recognizers which supports the supplied entities
         and language
         """
-        if language is None and entities is None:
-            return self.recognizers
-
         if language is None:
             raise ValueError("No language provided")
 
-        if entities is None:
+        if entities is None and all_fields is False:
             raise ValueError("No entities provided")
 
         to_return = []
-        for entity in entities:
-            subset = [rec for rec in self.recognizers if
-                      entity in rec.supported_entities
-                      and language == rec.supported_language]
-
-            if len(subset) == 0:
-                logging.warning(
-                    "Entity " + entity +
-                    " doesn't have the corresponding recognizer in language :"
-                    + language)
-            else:
-                to_return.extend(subset)
+        if all_fields:
+            to_return = [rec for rec in self.recognizers if
+                         language == rec.supported_language]
+        else:
+            for entity in entities:
+                subset = [rec for rec in self.recognizers if
+                          entity in rec.supported_entities
+                          and language == rec.supported_language]
+
+                if len(subset) == 0:
+                    logging.warning(
```

Comment:
this line is going to fail linting once the other lint pr is merged .
go for:
"text text %s text", "text"
style